### ISSUE TITLE
custom wasm file fails to run

### ISSUE THREAD

**Issue by naps62** [DelphinusLab/zkWasm#69]
I'm trying to write a Hello world kind of program. Suposedly, I should be able to compile a wasm file and run it through zkWasm.

But I'm failing to understand how exactly my wasm should be compiled. Or at least I'm assuming there's some discrepancy there that leads to the run failing

I first ran `test_all.sh` which relies on the existing `bsearch_64.wasm`. There's a C version, so I went from there.

Here's my code:

```c
// comp.c
unsigned long long wasm_input(int);

unsigned int main() {
  unsigned long long x = wasm_input(1);

  return x > 4 ? 1 : 0;
}
```

```Justfile
FLAGS := "-flto -O3 -nostdlib -fno-builtin -ffreestanding -mexec-model=reactor --target=wasm32 -Wl,--strip-all -Wl,--no-entry -Wl,--allow-undefined -Wl,--compress-relocations -Wl,--export-dynamic"

cli := "../../DelphinusLab/zkWasm/target/release/cli"

build:
  clang {{FLAGS}} -Wl,--export=main -o comp.wasm comp.c

run:
  just build

  {{cli}} \
    --function main \
    --output ./output \
    --wasm ./comp.wasm \
    setup

  {{cli}} \
    --function main \
    --output ./output \
    --wasm ./comp.wasm \
    single-prove \
    --public 5:i64
```

the `run` task fails to execute:

```
❯ j run
just build
clang -flto -O3 -nostdlib -fno-builtin -ffreestanding -mexec-model=reactor --target=wasm32 -Wl,--strip-all -Wl,--no-en
try -Wl,--allow-undefined -Wl,--compress-relocations -Wl,--export-dynamic -Wl,--export=main -o comp.wasm comp.c
../../DelphinusLab/zkWasm/target/release/cli --function main --output ./output --wasm ./comp.wasm setup
read params K=18 from "./output/K18.params"
read vkey from "./output/zkwasm.0.vkey.data"
quotient_poly_degree 8
read params K=23 from "./output/K23.params"
../../DelphinusLab/zkWasm/target/release/cli --function main --output ./output --wasm ./comp.wasm single-prove --publi
c 5:i64
thread 'main' panicked at 'failed to execute export: Trap(Code(UnexpectedSignature))', src/runtime/wasmi_interpreter.r
s:93:14
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
error: Recipe `run` failed on line 22 with exit code 101
```

**Comment by xgaozoyoe** [comment]
Thanks, I am analyzing it.

**Comment by junyu0312** [comment]
Please rename 'main' function since it contains two arguments. You may check it via
```
wasm2wat comp.wasm
```

BTW, Passing parameters to top-level functions is not supported; instead, use
`wasm_input(0)` and `wasm_input(1)` to pass private and public arguments, respectively.

**Comment by naps62** [comment]
@junyu0312 I think I got it working, thanks.
But I hadn't realized before that `wasm_input` was both for public and private inputs. is there some kind of documentation I could read to get started? I guess I'm missing a bunch of other things

**Comment by xgaozoyoe** [comment]
Closed this and rely on document issue #40 to track this thread.
